src/herder/Upgrades.cpp

```diff
@@ -1466,42 +1466,122 @@ ConfigUpgradeSetFrame::isValidForApply() const
+                        MinimumSorobanNetworkConfig::MIN_READ_BYTES &&
+                    cfg.contractLedgerCost().ledgerMaxWriteLedgerEntries >
+                        MinimumSorobanNetworkConfig::MIN_WRITE_ENTRIES &&
+                    cfg.contractLedgerCost().ledgerMaxWriteBytes >
+                        MinimumSorobanNetworkConfig::MIN_WRITE_BYTES &&
+                    cfg.contractLedgerCost().txMaxReadLedgerEntries >
+                        MinimumSorobanNetworkConfig::MIN_READ_ENTRIES &&
+                    cfg.contractLedgerCost().txMaxReadBytes >
+                        MinimumSorobanNetworkConfig::MIN_READ_BYTES &&
+                    cfg.contractLedgerCost().txMaxWriteLedgerEntries >
+                        MinimumSorobanNetworkConfig::MIN_WRITE_ENTRIES &&
+                    cfg.contractLedgerCost().txMaxWriteBytes >
+                        MinimumSorobanNetworkConfig::MIN_WRITE_BYTES &&
+                    cfg.contractLedgerCost().feeReadLedgerEntry > 0 &&
+                    cfg.contractLedgerCost().feeWriteLedgerEntry > 0 &&
+                    cfg.contractLedgerCost().feeRead1KB > 0 &&
+                    cfg.contractLedgerCost().bucketListTargetSizeBytes > 0 &&
+                    cfg.contractLedgerCost().writeFee1KBBucketListLow > 0 &&
+                    cfg.contractLedgerCost().writeFee1KBBucketListHigh > 0 &&
+                    cfg.contractLedgerCost().bucketListWriteFeeGrowthFactor > 0;
+
+            valid =
+                valid && cfg.contractLedgerCost().ledgerMaxReadLedgerEntries >=
+                             cfg.contractLedgerCost().txMaxReadLedgerEntries;
+            valid = valid && cfg.contractLedgerCost().ledgerMaxReadBytes >=
+                                 cfg.contractLedgerCost().txMaxReadBytes;
+
+            valid =
+                valid && cfg.contractLedgerCost().ledgerMaxWriteLedgerEntries >=
+                             cfg.contractLedgerCost().txMaxWriteLedgerEntries;
+            valid = valid && cfg.contractLedgerCost().ledgerMaxWriteBytes >=
+                                 cfg.contractLedgerCost().txMaxWriteBytes;
+            break;
         case ConfigSettingID::CONFIG_SETTING_CONTRACT_META_DATA_V0:
+            valid = cfg.contractMetaData().txMaxExtendedMetaDataSizeBytes >
+                        MinimumSorobanNetworkConfig::MIN_WRITE_BYTES &&
+                    cfg.contractMetaData().feeExtendedMetaData1KB > 0;
+            break;
         case ConfigSettingID::CONFIG_SETTING_STATE_EXPIRATION:
-            // For now none of these settings have any semantical value.
-            // Validation should be implemented when implementing/tuning
-            // the respective settings.
-            valid = true;
+            valid =
+                cfg.stateExpirationSettings().maxEntryExpiration > 0 &&
+                cfg.stateExpirationSettings().minTempEntryExpiration > 0 &&
+                cfg.stateExpirationSettings().minPersistentEntryExpiration >
```

Comment:
I think we need a lower bound on max entry expiration in order to make sure that we can bump the upgrade entry reasonably (a few days/weeks?). Also probably need some small threshold on min persistent expiration, like 100 ledgers or something, so that there is time to bump it.